### PR TITLE
add glm ppo

### PR DESCRIPTION
**Pull request by ygdl0228**

### CODE REVIEW

File: examples/glm4_train_tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# model:       模型名称
+# input_path:  下载HuggingFace权重的文件夹路径，注意最后面有/
+# output_path: 转换后的MindSpore权重文件保存路径
+# dtype:       转换权重的精度
+```
+
+### 推荐依赖版本
+
+mindformers == dev
+mindspore == 2.4
```
Comment (ChessQian) [inline examples/glm4_train_tutorial/README.md:31]: 2.4.0还是2.4.1，需注明

Comment (ChessQian) [reply examples/glm4_train_tutorial/README.md:31]: 或者直接2.4.1测试一下

File: mindrlhf/models/ppo_models.py [hunk 2]
```diff
@@ -285,6 +291,7 @@ def construct(self,
         if samples is not None:
             logprobs_labels = self.logprobs_of_labels(logits, samples, batch_size, seq_length)
             if return_value == True:
+                # OG code is not equal to the logic in criticmodel
```
Comment (ChessQian) [inline mindrlhf/models/ppo_models.py:294]: OG是啥？

Comment (ChessQian) [reply mindrlhf/models/ppo_models.py:294]: 不需要的注释不要加

File: model_configs/glm4_config/finetune_glm4_9b.yaml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  loss_repeated_mean: True
+  enable_alltoall: False
+  full_batch: True   # 默认为True, 数据并行模式必须设置为False
+  search_mode: "sharding_propagation"
+  enable_parallel_optimizer: True  # optimizer shard
+  strategy_ckpt_config:
+    save_file: "./ckpt_strategy.ckpt"
+parallel_config:
+  data_parallel: 1
+  model_parallel: 4
```
Comment (ChessQian) [inline model_configs/glm4_config/finetune_glm4_9b.yaml:160]: 不设置成8进行finetune么

File: examples/glm4_train_tutorial/README.md (hunk 1 above)
Comment (ygdl0228) [reply examples/glm4_train_tutorial/README.md]: 已经改为2.4.1

File: mindrlhf/models/ppo_models.py (hunk 2 above)
Comment (ygdl0228) [reply mindrlhf/models/ppo_models.py]: 已删去注释

File: model_configs/glm4_config/finetune_glm4_9b.yaml (hunk 3 above)
Comment (ygdl0228) [reply model_configs/glm4_config/finetune_glm4_9b.yaml:160]: mp8报错 所以暂时mp4
